code/parachain/frame/staking-rewards/src/lib.rs

```diff
@@ -918,111 +922,100 @@ pub mod pallet {
 			Ok(())
 		}
 
+		// TODO(benluelo): Split this out into a separate function/file
 		#[transactional]
 		fn split(
 			who: &Self::AccountId,
-			(fnft_collection_id, fnft_instance_id): &Self::PositionId,
+			(fnft_collection_id, existing_fnft_instance_id): &Self::PositionId,
 			ratio: Permill,
-		) -> Result<[Self::PositionId; 2], DispatchError> {
-			// TODO(benluelo): Refactor
-			let keep_alive = false;
-			let mut old_position_stake = BalanceOf::<T>::zero();
-			let mut old_position =
-				Stakes::<T>::try_mutate(fnft_collection_id, fnft_instance_id, |maybe_stake| {
-					let stake = maybe_stake.as_mut().ok_or(Error::<T>::StakeNotFound)?;
-
-					let old_value = stake.clone();
-					stake.stake = ratio.mul_floor(stake.stake);
-					stake.share = ratio.mul_floor(stake.share);
+		) -> Result<Self::PositionId, DispatchError> {
+			let (new_fnft_instance_id, new_position) = Stakes::<T>::try_mutate(
```

Comment:
I don't see a value in changing the structure here, the previous code was better

Reply:
I'd argue this implementation looks cleaner. However, I'm not sure which one is more correct in terms of what this function returns. Who's the end user of these return values, and what do they want from the values returned?